src/components/CityCard.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+import { observer } from "mobx-react-lite";
+import { useStore } from "stores";
+import { CityWithUTCOffset } from 'types';
+import './CityCard.scss';
+
+function formatAMPM(date: Date) {
+    let hours = date.getHours();
+    const minutes = date.getMinutes();
+    const ampm = hours >= 12 ? 'pm' : 'am';
+    hours = hours % 12;
+    hours = hours ? hours : 12;
+    const minutesStr = minutes < 10 ? `0${minutes}` : `${minutes}`;
+    const strTime = `${hours}:${minutesStr} ${ampm}`;
+    return strTime;
+}
+
+interface CityCardProps {
+    city?: CityWithUTCOffset;
+}
+
+export const CityCard = observer(({ city }: CityCardProps) => {
+    const { appStore } = useStore();
+
+    const time = useMemo(() => {
+        const date = new Date(appStore.timer + (city ? city.utcOffset * 60 * 1000 : 0));
```

Comment:
You can easily format time using modern native date methods and avoid all boilerplate related to utcOffset property and am/pm formatting